### PR TITLE
Created place and distance microservices

### PR DESCRIPTION
**Pull request by EquablePanic4**
Created two independent microservices - place and distance. Change design pattern.

### CODE REVIEW

**Review by rogowski-piotr — CHANGES_REQUESTED** [review]

File: .NET/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,25 @@
+**/.classpath
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/.dockerignore:1]: Good point - this file doesn't do anything in that place. It should be moved to project's directories.

**Review by rogowski-piotr — CHANGES_REQUESTED** [review]
We added e2e tests please update your branch and uncomment [these](https://github.com/rogowski-piotr/system-benchmarking-monolith-vs-microservices/blob/8ab583c8efe8d2252659e3864936e20bb68cac98/.github/workflows/test.yml#L88) lines to test your changes 😉

File: .NET/Microservices/DistanceService/DistanceService.csproj.user [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <ActiveDebugProfile>Docker</ActiveDebugProfile>
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/Microservices/DistanceService/DistanceService.csproj.user:4]: This is just Visual Studio file

File: .NET/Microservices/PlaceService/Program.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+using PlaceService.Repositories;
+using PlaceService.Responses;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+var repository = new PlaceRepository();
+
+//Get all places
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/Microservices/PlaceService/Program.cs:9]: Yes, there is possibility to make separate controllers

File: .NET/SystemBenchmarkingMonolithVsMicroservices.sln [hunk 4]
```diff
@@ -0,0 +1,68 @@
+﻿
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/SystemBenchmarkingMonolithVsMicroservices.sln:1]: Yes, we really need this file, it's Visual Studio Solution file.

File: .NET/Microservices/DistanceService/Requests/DistanceRequest.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+        }
+
+        private double ParseToLatitude(string cord)
+            => Double.Parse(regex.Match(cord).Groups[1].Value, CultureInfo.InvariantCulture);
+
+        private double ParseToLongitude(string cord)
+            => Double.Parse(regex.Match(cord).Groups[2].Value, CultureInfo.InvariantCulture);
+
+        public List<Point> GetPoints()
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/Microservices/DistanceService/Requests/DistanceRequest.cs:33]: Yes, it is.

File: .NET/Microservices/DistanceService/Program.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+app.MapPost("/api/distance", (DistanceRequest req) =>
+{
+    var points = req.GetPoints();
+
+    var point1 = points[0];
+    var point2 = points[1];
+
+    var distance = distanceService.CalculateDistance(point1.Latitude, point1.Longitude, point2.Latitude, point2.Longitude);
+
+    return Results.Ok(new DistanceResponse(distance));
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/Microservices/DistanceService/Program.cs:21]: No problem

File: .NET/Microservices/DistanceService/Services/SphericalDistanceService.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    public class SphericalDistanceService
+    {
+        private const double RADIANS_MULTIPLICATION = Math.PI / 180.00;
+
+        public double CalculateDistance(double lat1, double long1, double lat2, double long2)
+            => Math.Acos(Math.Sin(ToRadians(lat1)) * Math.Sin(ToRadians(lat2)) +
+                Math.Cos(ToRadians(lat1)) * Math.Cos(ToRadians(lat2)) *
+                Math.Cos(ToRadians(long1 - long2))) * 6371;
+
+        //public double CalculateDistance(double lat1, double long1, double lat2, double long2)
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/Microservices/DistanceService/Services/SphericalDistanceService.cs:12]: You're right

File: .NET/Microservices/PlaceService/Models/Place.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Text.Json.Serialization;
+
+namespace PlaceService.Models
+{
+    public class Place
+    {
+        [JsonPropertyName("id")]
```
[the remark this answers is not in the document]
Comment (EquablePanic4) [reply .NET/Microservices/PlaceService/Models/Place.cs:7]: [JsonPropertyName] is an attribute which makes Json Serializer use exactly that name while serializing or deserializing objects

**Review by rogowski-piotr — APPROVED** [review]
Good Job 🥇 Thanks!

**Pull request merged**